### DISCUSSION TITLE
frontend-host options

### DISCUSSION THREAD

**Discussion by makerjackie** [reflex-dev/reflex#3128] [Q&A] [answered]
I need to debug via a remote machine, but how can I change the port for a frontend webpage? I am currently able to access the API documentation via http://ip:8000/docs, but I cannot access the website through http://ip:3000."

I couldn't find a `frontend-host` option when run `reflex run`

```
Usage: reflex run [OPTIONS]

  Run the app in the current directory.

Options:
  --env [dev|prod]                The environment to run the app in.
                                  [default: dev]
  --frontend-only                 Execute only frontend.
  --backend-only                  Execute only backend.
  --frontend-port TEXT            Specify a different frontend port.
                                  [default: 3000]
  --backend-port TEXT             Specify a different backend port.  [default:
                                  8000]
  --backend-host TEXT             Specify the backend host.  [default:
                                  0.0.0.0]
  --loglevel [debug|info|warning|error|critical]
                                  The log level to use.  [default: info]
  --help                          Show this message and exit.
```

**Comment by picklelo** [comment]
You can set the `--frontend-port` flag to change it (or set the `FRONTEND_PORT` environment variable).

**Reply by makerjackie** [reply]
I want to change frontend-host to 0.0.0.0 instead of localhost, not port

**Comment by makerjackie** [comment]
according to docs: https://reflex.dev/docs/api-reference/config/

change rxconfig.py 
```python
config = rx.Config(
    app_name="blank",
    frontend_port=8001,
    deploy_url="http://0.0.0.0:8001/",
)
```

But add a option `frontend-host` like `backend-host` to `reflex run` will be better @picklelo @masenf

**Comment by masenf** [comment, marked as answer]
I think i'm still a bit confused by this discussion... the `deploy_url` _only_ affects the hosting service and the sitemap generation for export. From double checking the code, `deploy_url` seems to change nothing about how `reflex run` serves the site.

By default, the frontend should be available from other hosts without further configuration.

**Reply by makerjackie** [reply]
Sorry, it's my fault, and you are right. The first loading of the website took a long time, and I thought it was inaccessible. In addition, initially, my inability to access it might be due to the fact that the firewall did not open the port.

**Reply by makerjackie** [reply]
and from `reflex run --loglevel debug`, I saw
```
Debug:    - Local:        http://localhost:3000
App running at: http://localhost:3000
```
and I thought it was inaccessible at 0.0.0.0

**Reply by masenf** [reply]
I think the docs probably could be more clear on what deploy_url is actually for though. And definitely the console output suggests that Reflex is only listening on localhost.

**Reply by makerjackie** [reply]
Although the console prompts that it can only be accessed at [localhost](https://localhost/), I can indeed access it through an external IP. In addition, the first access on my computer waited for about 7 seconds.

**Reply by jakeayres** [reply]
It doesn't appear to me that `deploy_url="http://0.0.0.0:10000` does set the host of the frontend. I'm attempting to deploy a Reflex app to Render.com and their port scanner is unable to detect the frontend on 0.0.0.0 but is able to detect the backend.

This is proving to be a critical issue for anyone intending to deploy to Render.

**Reply by masenf** [reply]
@jakeayres with render, only a single port can be exposed, so my example [here](https://github.com/masenf/render-reflex-webcam/) uses a reverse proxy (caddy) to route the traffic to the frontend or backend by url while exposing a single port, 10000

**Reply by jakeayres** [reply]
I'm running my app across two render services (backend and frontend). The backend appears to be working perfectly well as I can interface with it from a locally running frontend (`--frontend-only`). Despite my best efforts, I can't convince render to see my remote frontend and expose it to the public at `<myappname>.onrender.com`. The logs indicate that the frontend is running at localhost:10000 and that render can't find any ports on 0.0.0.0 even with `deploy_url="http://0.0.0.0:10000",` in my config.

**Reply by masenf** [reply]
the `deploy_url` probably isn't doing what you want.

on the frontend try setting `api_url="https://mybackend.onrender.com"`

**Reply by jakeayres** [reply]
My full config is:

```
config = rx.Config(
    app_name="mybackend",
    api_url="https://mybackend.onrender.com",
    frontend_port=10000,
    deploy_url="http://0.0.0.0:10000",
)
```

My backend service is run remote on render using `reflex run --backend-only --backend-port 10000`

If I run a _local_ front end using `reflex run --frontend-only` all is well. Indeed, if I suspend the backend service on render, I am notified that my local front end is no longer able to connect to the backend.

If i run a _remote_ front end on render with start command `reflex run --frontend-only --frontend-port 10000`. The logs return:

`App running at: http://localhost:10000/ (Frontend-only mode)`
`==> No open HTTP ports detected on 0.0.0.0, continuing to scan...`

The service running the front end also exceeds its memory limit (2GB) if left alone for 10mins. It isn't clear to me whether this is related or not.

**Reply by jakeayres** [reply]
For what it is worth, I never managed to find a fix for running the app across two render services.

Dockerizing my app and using Caddy and running on one services (as you suggested) worked well.

**Reply by miketech** [reply]
> For what it is worth, I never managed to find a fix for running the app across two render services.
> 
> Dockerizing my app and using Caddy and running on one services (as you suggested) worked well.

Hi, I did it that way: Using Render Webservices to host the backend (backend-only) and Render Static Site to host a Reflex Export (frontend-only). Maybe this helps.
